### DISCUSSION TITLE
Options with platform specific default values

### DISCUSSION THREAD

**Discussion by ivwumupy** [xmake-io/xmake#5839] [Q&A]
Hi! I'm trying to add an option whose default value depends on the platform. The following doesn't seem to work
```lua
option("enable_metal", function ()
  if is_plat("macosx") then
    set_default(true)
  else
    set_default(false)
  end
  add_defines("ENABLE_METAL")
end)
target("my_target", function ()
  add_options("enable_metal")
  -- ...
end)
```

**Comment by waruqi** [comment]
see https://github.com/tboox/tbox/blob/3fdd8f0ab91cea9fe516ab31035ea79425c04c4a/src/xmake.lua#L70

https://github.com/tboox/tbox/blob/3fdd8f0ab91cea9fe516ab31035ea79425c04c4a/src/xmake.lua#L27
